OfficeDev/office-ui-fabric-react
# Convert ContextualMenu and ContextualMenuItem to CSS in JS

#### Pull request checklist

Marked as **minor** bump.

- [x] Addresses an existing issue: Fixes #4169

#### Description of changes

This PR (hopefully) finishes the refactor of `ContextualMenu` and `ContextualMenuItem` to use the CSS in JS approach of `mergeStyles` building off of #5356 and #5461. 

**List of Changes:**

- Move interfaces to their appropriate `*.types.ts` file for consistency i.e. `IContextualMenuClassNames` now resides in `ContextualMenuItem.types.ts`.
- Exports `IContextualMenuClassNames ` and `IMenuItemClassNames` as part of fixing #4169 and marks them as `@deprecated`.
- Mark `get*ClassNames` methods as `@deprecated`.
- Condense the `getStyles` call for each component into a single method per component.
- `ContextualMenuItemBase` item render functions refactored from `static` to `private` instance methods to access processed class names.
- Use `getGlobalClassNames` for `getStyle` methods.
- Deprecate in-line `IContextualMenuItem `'s `style` prop usage and suggest `mergeStyles` API.
- Refactor redundant `IMenuItemStyles` into `IContextualMenuItemStyles`.
- Remove unused & deprecated test files.

#### Focus areas to test

- Make sure `ContextualMenu` examples render properly.
- Check public API surface particularly backwards compatibility
- Try customizing `ContextualMenu` and `ContextualMenuItem` components using `styles` props.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/5885)
